### ISSUE TITLE
Unable to mount nfs shares as user in dolphin using KDE

### ISSUE THREAD

**Issue by johnramsden** [NixOS/nixpkgs#24913]
## Issue description

I have setup NFS mounts that should be available to users but I am unable to mount them in dolphin. I have installed `nixos.kdeApplications.dolphin-plugins` and `nixos.kdeApplications.kdenetwork-filesharing` as well as `nfs-utils`. I'm able to visit the share as my user from the command line, and as soon as I have done that the shares are available in dolphin. The problem is after boot they're unavailable until I have visited the share from the command line. I know it is possible to have this working since I have the exact same setup working properly on Arch.

When visited from dolphin I get the message:

> An error occurred while accessing '/mnt/tank/media/Downloads/Incomplete on lilan.ramsden.network', the system responded: mount.nfs: not installed setuid - "user" NFS mounts not supported.

I have tried nfsv4, user mount option and users mount option. All give a similar error.

### Steps to reproduce

Attempt to reach user mountable shares from dolphin.

## Technical details

My mount setup:

```
  fileSystems."/mnt/lilan/media/Downloading/Downloads/Incomplete" =
    { device = "lilan.ramsden.network:/mnt/tank/media/Downloads/Incomplete";
      fsType = "nfs";
      options = [ "noauto" "x-systemd.automount" "x-systemd.device-timeout=175" "timeo=15" "x-systemd.idle-timeout=1min" "user" ];
    };
```

* System: (NixOS: 17.03.928.69d9061908 (Gorilla))
* Nix version: (nix-env (Nix) 1.11.8)
* Nixpkgs version: ("17.03.928.69d9061908")

**Comment by edanaher** [comment]
I think the issue here is that mount isn't installed setuid; the Linux standard is that mount is setuid root, and then mount enforces permissions itself by only allowing non-root users to mount filesystems that are defined in /etc/fstab with the user option.  But Nix doesn't do that, so anything not running as root just can't mount filesystems.

I should try to figure out if there's a reason why Nix is nonstandard here.

In the meantime, you can add setuid mount/umount wrappers by putting the following in configuration.nix:
```nix
  security.wrappers = {
    mount.source = "${pkgs.utillinux}/bin/mount";
    umount.source = "${pkgs.utillinux}/bin/umount";
  };
```

**Comment by edanaher** [comment]
And a few minutes of research turned up #9848.  My summary of that discussion is that setuid programs are a Bad Idea in general, and there's a more modern approach to handing user mounts (udisks) that handles things better.

Some quick googling also suggests that dolphin should support udisks; there might be some configuration Nix should be doing to enable that.  I don't know if I'll have time to look into that in the next few days.

**Comment by johnramsden** [comment]
@edanaher I'm familiar with udisks, guess i'll try to set it up.

**Comment by ttuegel** [comment]
By the way, `dolphin-plugins` is part of `dolphin` by default; installing it doesn't do anything.

**Comment by nioncode** [comment]
Setting the security.wrappers for mount doesn't work for me using a cifs mount.
When I do a `mount /mnt/share` as a regular user, I get: `mount: /mnt/share: operation permitted for root only`.

**Comment by dschrempf** [comment]
Hi, I just tried the `security.wrapper` option, and it does not work. I still get the same error:
```
mount.nfs: not installed setuid - "user" NFS mounts not supported.
```

**Comment by ccressent** [comment]
@dschrempf 
`mount.nfs` is a different binary than `mount` and `umount` and is provided by `pkgs.nfs-utils`. You might want to add this to your configuration:

```
  security.wrappers."mount.nfs".source = "${pkgs.nfs-utils.out}/bin/mount.nfs";
```

**Comment by dschrempf** [comment]
Thanks. That's what I am actually using at the moment. I must have commented here before finding the solution... Thank you!

**Comment by happyalu** [comment]
While updating to 21.11, got an error: `error: The option `security.wrappers.mount.nfs.owner' is used but not defined.`.

Had to explicitly set the fields to get it to work:

```
  security.wrappers."mount.nfs" = {
    setuid = true;
    owner = "root";
    group = "root";
    source = "${pkgs.nfs-utils.out}/bin/mount.nfs";
  };
```

**Comment by dschrempf** [comment]
Yes, I am also setting the `owner`, and the `group`. This was a change introduced some time ago.
